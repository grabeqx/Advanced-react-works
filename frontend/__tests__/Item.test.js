import ItemComponent from '../components/Item';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

const fakeItem = {
  id: 'ABS',
  title: 'A cool Item',
  price: 400,
  description: 'lorem ipusm',
  image: 'image.png',
  largeImage: 'fullImage.png'
};

describe('<Item />', () => {
  it('renders and matches the snapshot', () => {
    const wrapper = shallow(<ItemComponent item={fakeItem} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  })

  // it('renders and displays properly', () => {
  //   const wrapper = shallow(<ItemComponent item={fakeItem} />);

  //   const priceTag = wrapper.find('PriceTag');

  //   console.log(wrapper.debug());
  //   expect(priceTag.children().text()).toBe('$5');
  //   expect(wrapper.find('Title a').text()).toBe(fakeItem.title);
  //   const img = wrapper.find('img');
  //   expect(img.props().src).toBe(fakeItem.image);
  //   expect(img.props().alt).toBe(fakeItem.title);
  // });

  // it('renders out the buttons properly', () => {
  //   const wrapper = shallow(<ItemComponent item={fakeItem} />);
  //   const buttonList = wrapper.find('.buttonList');
  //   expect(buttonList.children()).toHaveLength(3);
  //   expect(buttonList.find('Link')).toHaveLength(1);
  //   expect(buttonList.find('Link')).toBeTruthy();
  //   expect(buttonList.find('AddToCart')).toBeTruthy();
  //   expect(buttonList.find('DeleteButton')).toBeTruthy();

  // });

})