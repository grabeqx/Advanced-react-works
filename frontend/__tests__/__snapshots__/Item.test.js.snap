// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item /> renders and matches the snapshot 1`] = `
<ItemStyles__Item>
  <img
    alt="A cool Item"
    src="image.png"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "ABS",
          },
        }
      }
    >
      <a>
        A cool Item
      </a>
    </Link>
  </Title>
  <PriceTag>
    $4
  </PriceTag>
  <p>
    lorem ipusm
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "update",
          "query": Object {
            "id": "ABS",
          },
        }
      }
    >
      <a>
        Edit
      </a>
    </Link>
    <AddToCart
      id="ABS"
    />
    <DeleteButton
      id="ABS"
    >
      Delete this item
    </DeleteButton>
  </div>
</ItemStyles__Item>
`;
